contacts/main.js

```diff
@@ -0,0 +1,553 @@
+  minusButton.innerHTML = '-';
+  minusButton.custom = 'minus-button-custom';
+  if (newPerson.likes === 0)
+    minusButton.disabled = true;
+  newLi.appendChild(minusButton);
+
+  let image = document.createElement('img');
+  image.src = newPerson.image;
+  image.alt = 'pic';
+  image.className = 'avatar';
+  newLi.appendChild(image);
+
+  let checkbox = document.createElement('input');
+  checkbox.type = 'checkbox';
+  checkbox.custom = 'checkbox-custom';
+  checkbox.id = 'checkbox-' + newPerson.id;
+  newLi.appendChild(checkbox);
+
+  let full_name = document.createElement('span');
+  full_name.className = 'full-name';
+  let first_name = document.createElement('span');
+  first_name.innerHTML = newPerson.firstName;
+  first_name.style.marginRight = '10px';
+  full_name.appendChild(first_name);
+  let last_name = document.createElement('span');
+  last_name.innerHTML = newPerson.lastName;
+  full_name.appendChild(last_name);
+  newLi.appendChild(full_name);
+
+  let trash = document.createElement('img');
+  trash.id = 'trash-' + newPerson.id;
+  trash.src = 'images/trash-1.png';
+  trash.alt = 'trash';
+  trash.custom = 'trash-custom';
+  trash.className = 'trash';
+  newLi.appendChild(trash);
+
+  ul.appendChild(newLi);
+}
+
+/////////////////////handle for the contacts list///////////////////
+
+function eventsHandlerContacts(event) {
+  let target_id = event.target.id;
+  let personId = getIdOfPerson(target_id);
+  switch (event.target.custom) {
+    case 'minus-button-custom':
+      decrementLikes(personId);
+      refreshFavsList();
+      rerenderContacts();
+      break;
+    case 'plus-button-custom':
+      incrementLikes(personId);
+      refreshFavsList();
+      rerenderContacts();
+      break;
+    case 'trash-custom':
+      if (confirm('ARE YOU SURE YOU WANT TO DELETE THIS CONTACT')) {
+        deletePerson(personId);
+        refreshFavsList();
+        rerenderContacts();
+      }
+      break;
+    case 'checkbox-custom':
+      changeDeleteAllButtonValueAndState();
+      break;
+  }
+}
+
+///////////////////////////////////////////////////////////////////
+
+function changeDeleteAllButtonValueAndState() {
+  let button = document.getElementById('delete-all-button');
+  let numberOfCheckedContacts = 0;
+  numberOfCheckedContacts = getSelectedContactsIDs().length;
+  if (numberOfCheckedContacts !== 0) {
+    button.innerHTML = 'Delete All ' + numberOfCheckedContacts.toString();
+    button.disabled = false;
+  } else {
+    button.innerHTML = 'Delete All';
+    button.disabled = true;
+  }
+}
+
+function resetDeleteAllButtonValueAndState() {
+  document.getElementById('delete-all-button').innerHTML = 'Delete All';
+  document.getElementById('delete-all-button').disabled = true;
+}
+
+//////////////////////////////////////////////////////////////////
+
+function deletePerson(personId) {
+  for (let i = 0; i < contactsList.length; i++)
+    if (contactsList[i].id === personId) {
```

Comment:
instead of this use 
var index = contactsList.map(x => {
  return x.Id;
}).indexOf(id);
and then call 
contactsList.splice(index, 1);

this requires no for statement